File: app/src/java/java/com/example/ama/viewer/di/modules/DbModule.java
package com.example.ama.viewer.di.modules;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.realm.RealmConfiguration;

@Module
public class DbModule {

    @Provides
    @Singleton
    RealmConfiguration provideRealmConfiguration() {
        return new RealmConfiguration.Builder().build();
    }
}

File: app/src/kotlin/java/com/example/ama/viewer/di/modules/DbModule.kt
package com.example.ama.viewer.di.modules

import dagger.Module
import dagger.Provides
import io.realm.RealmConfiguration
import javax.inject.Singleton

@Module
class DbModule {

    @Provides
    @Singleton
    fun provideRealmConfiguration(): RealmConfiguration =
            RealmConfiguration.Builder().build()
}